File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Conne/CartColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CartRow } from "./CartRow";

export interface CartColumns {
    Id: Column<CartRow>;
    Userid: Column<CartRow>;
    ProductidName: Column<CartRow>;
    Count: Column<CartRow>;
    Price: Column<CartRow>;
}

export class CartColumns extends ColumnsBase<CartRow> {
    static readonly columnsKey = 'Conne.Cart';
    static readonly Fields = fieldsProxy<CartColumns>();
}

File: AdminPanel/AdminPanel.Web/Modules/Conne/Cart/CartColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel.Conne.Columns;

[ColumnsScript("Conne.Cart")]
[BasedOnRow(typeof(CartRow), CheckNames = true)]
public class CartColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Userid { get; set; }
    public string ProductidName { get; set; }
    public int Count { get; set; }
    public int Price { get; set; }
}